jep/400/README.adoc

```diff
@@ -0,0 +1,224 @@
+
+This JEP is a “stake on the ground” that captures the mission and the general directions.
+
+== Specification
+
+
+The Jenkins X sub project focuses 100% on Kubernetes, CI/CD and Cloud Native use cases to provide great developer productivity. To that end, we have so far identified the following major deliverables.
+
+=== Development Practice
+
+Jenkins X promotes a particular Git branching & repository model. The rest of the tools and services are built to fit this model.
+
+This development practice represents the best practice of developing Kubernetes applications, based in part on the experience of developing Fabric8, a project of a similar mission and on the results of the https://puppet.com/resources/whitepaper/state-of-devops-report[State of DevOps report].
+
+The value of Jenkins X is that if you follow this best practice, then Jenkins X assembles all the pieces for you (e.g., Jenkins, Kubernetes, Git, CI/CD etc) such that developers can be instantly productive. This is akin to what Maven brought over Ant —  encouraging developers to use a standard Lifecycle-model provided by Maven (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself[DRY]) to achieve greater productivity.
+
+Here are some examples of the practice:
+
+* The master branch shall be always clean and release ready. No long running feature branches to stay lean.
+* Pull requests are used to work on new changes that then land to the master. CI tests are triggered on changes to a PR and the PR is only merged if all the CI checks pass.
+* A release is produced from the master branch to create versioned immutable artifacts (jars, binaries, docker images, helm charts etc). A release might be triggered manually, or automatically upon a new PR merge, or even at set frequency.
+* Which version of what services run in which environment is managed declaratively in a separate environment Git repo. You make deployment happen by pushing changes into the environment git repository which then triggers the Environments pipeline. This approach is often referred to as _GitOps_ and is similar to how people develop Chef recipes or Ansible playbooks with Git.
+
+=== ‘Environment’ Model
+
+Among web application developers practicing Continuous Delivery, the notion of “environment” is a well-established practice. It allows orderly flow of changes from a developer, through testing and staging to production. Yet, Jenkins does not have the environment concept as the first class citizen.
+
+Jenkins X bridges this gap by introducing the concept of ‘environment’ on top of more generic Kubernetes concepts, such as namespaces and GitOps. The development practice then talks in terms of promoting apps from one environment to another in a cascading manner.
+
+Promotion of a new version to an Environment in the Jenkins X GitOps model results in
```

Comment:
I used `promotion` instead of `releasing` as they are separate operations. e.g. 1.2.3 is released as immutable artefacts (docker images, helm charts, jars etc).  Then the version of the app gets promoted from 1.2.2 to 1.2.3 by a PR on the environment's git repo. Lemme try polish that section to make it more clear